### ISSUE TITLE
[DOXIA-583] XhtmlBaseSink: Table caption is escaped incorrectly

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-doxia#548]
**[Iwao AVE!](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ave)** opened **[DOXIA-583](https://issues.apache.org/jira/browse/DOXIA-583?redirect=false)** and commented

Background:

We use maven-site-plugin to generate our documentation site (source is xdoc) and noticed that non-ASCII characters like Japanese or Chinese in table caption are not correctly displayed in the output html files.
During generation, these characters are encoded to entities (e.g. '&amp;#x548c;' ) and are displayed correctly in the browser.
However, in a table caption, the first  '&amp;' is escaped into '&amp;amp;'.
So, for example, the actual output becomes '&amp;amp;#x548c;' while the expected output is '&amp;#x548c;'.

To verify the issue, modify org.apache.maven.doxia.sink.impl.XhtmlBaseSinkTest.testTableCaption() as follows.

```java
    public void testTableCaption()
    {
        try
        {
            sink = new XhtmlBaseSink( writer );

            sink.table();
            sink.tableRows( null, false );
            sink.tableCaption( attributes );
            // Insert '&'
            sink.text( "cap&tion" );
            sink.tableCaption_();
            sink.tableRows_();
            sink.table_();
        }
        finally
        {
            sink.close();
        }

        // the inserted '&' will become '&amp;amp;' instead of '&amp;'
        assertEquals( "<table border=\"0\" class=\"bodyTable\">" +
                "<caption style=\"bold\">cap&amp;tion</caption></table>", writer.toString() );
    }
```

Not sure if this is a proper fix, but I modified the following line in org.apache.maven.doxia.sink.impl.XhtmlBaseSink.write(String) ...

```java
this.tableCaptionXMLWriterStack.getLast().writeText( unifyEOLs( text ) );
```

... to ...

```java
this.tableCaptionXMLWriterStack.getLast().writeMarkup( unifyEOLs( text ) );
```

... and the issue was resolved without breaking existing tests.

I have attached the above modification as a patch.
Please let me know if you prefer a PR on GitHub or need more info.

Thanks in advance,
Iwao

---

**Affects:** 1.8

**Attachments:**
- [table-caption-escape.patch](https://issues.apache.org/jira/secure/attachment/12951901/table-caption-escape.patch) (_1.87 kB_)

**Remote Links:**
- [GitHub Pull Request #17
](https://github.com/apache/maven-doxia/pull/17)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Please do a GitHub PR, I will take a look for 1.9.

**Comment by jira-importer** [comment]
**[Iwao AVE!](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ave)** commented

That would be great, thanks!
Here is the PR : https://github.com/apache/maven-doxia/pull/17

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [1eea8214aced4fcad79c6e9b6dc44518777ab8e8](https://gitbox.apache.org/repos/asf?p=maven-doxia.git;a=commit;h=1eea8214aced4fcad79c6e9b6dc44518777ab8e8).
